Wreorder : Problem

**Describe the bug**
> Warn when the order of member initializers given in the code does not match the order in which they must be executed.

`man 1 gcc` (Included in `-Wall` for C++)

> ISO C++ requires field designators to be specified in declaration order; field B will be initialized after field A

https://clang.llvm.org/docs/DiagnosticsReference.html#wreorder

```
~/Detours/src/disasm.cpp: In constructor ‘CDetourDis::CDetourDis(BYTE**, LONG*)’:
~/Detours/src/disasm.cpp:301:25: warning: ‘CDetourDis::m_bF3’ will be initialized after [-Wreorder]
  301 |     BOOL                m_bF3; // x86 only
      |                         ^~~~~
~/Detours/src/disasm.cpp:298:25: warning:   ‘BOOL CDetourDis::m_bVex’ [-Wreorder]
  298 |     BOOL                m_bVex;
      |                         ^~~~~~
~/detours/disasm.cpp:326:1: warning:   when initialized here [-Wreorder]
  326 | CDetourDis::CDetourDis(_Out_opt_ PBYTE *ppbTarget, _Out_opt_ LONG *plExtra) :
      | ^~~~~~~~~~
In file included from ~/Detours/src/disolx86.cpp:2:
~/Detours/src/disasm.cpp: In constructor ‘CDetourDisX86::CDetourDisX86(BYTE**, LONG*)’:
~/Detours/src/disasm.cpp:301:25: warning: ‘CDetourDisX86::m_bF3’ will be initialized after [-Wreorder]
  301 |     BOOL                m_bF3; // x86 only
      |                         ^~~~~
~/Detours/src/disasm.cpp:298:25: warning:   ‘BOOL CDetourDisX86::m_bVex’ [-Wreorder]
  298 |     BOOL                m_bVex;
      |                         ^~~~~~
In file included from ~/Detours/src/disolx86.cpp:2:
~/Detours/src/disasm.cpp:45:33: warning:   when initialized here [-Wreorder]
   45 | #define CDetourDis              CDetourDisX86
      |                                 ^~~~~~~~~~~~~
~/Detours/src/disasm.cpp:45:33: note: in definition of macro ‘CDetourDis’
   45 | #define CDetourDis              CDetourDisX86
      |                                 ^~~~~~~~~~~~~
In file included from ~/Detours/src/disolx64.cpp:2:
~/Detours/src/disasm.cpp: In constructor ‘CDetourDisX64::CDetourDisX64(BYTE**, LONG*)’:
~/Detours/src/disasm.cpp:301:25: warning: ‘CDetourDisX64::m_bF3’ will be initialized after [-Wreorder]
  301 |     BOOL                m_bF3; // x86 only
      |                         ^~~~~
~/Detours/src/disasm.cpp:298:25: warning:   ‘BOOL CDetourDisX64::m_bVex’ [-Wreorder]
  298 |     BOOL                m_bVex;
      |                         ^~~~~~
In file included from ~/Detours/src/disolx64.cpp:2:
~/Detours/src/disasm.cpp:57:33: warning:   when initialized here [-Wreorder]
   57 | #define CDetourDis              CDetourDisX64
      |                                 ^~~~~~~~~~~~~
~/Detours/src/disasm.cpp:57:33: note: in definition of macro ‘CDetourDis’
   57 | #define CDetourDis              CDetourDisX64
      |
```

**Expected behavior**
Compiles without warnings

**Detours version**
https://github.com/microsoft/Detours/commit/827b89608ce66892963d1467d15a86f0b2832a78

--- a/src/creatwth.cpp
+++ b/src/creatwth.cpp
@@ -1436,7 +1436,7 @@ BOOL WINAPI DetourProcessViaHelperDllsW(_In_ DWORD dwTargetPid,
     PDETOUR_EXE_HELPER helper = NULL;
     HRESULT hr;
     WCHAR szDllName[MAX_PATH];
-    int cchWrittenWideChar;
+    unsigned int cchWrittenWideChar;
     DWORD nLen = GetEnvironmentVariableW(L"WINDIR", szExe, ARRAYSIZE(szExe));
 
     DETOUR_TRACE(("DetourProcessViaHelperDlls(pid=%lu,dlls=%lu)\n", dwTargetPid, nDlls));

--- a/src/detours.cpp
+++ b/src/detours.cpp
@@ -1466,7 +1466,7 @@ static BOOL detour_is_region_empty(PDETOUR_REGION pRegion)
 
     // Stop if any of the trampolines aren't free.
     PDETOUR_TRAMPOLINE pTrampoline = ((PDETOUR_TRAMPOLINE)pRegion) + 1;
-    for (int i = 0; i < DETOUR_TRAMPOLINES_PER_REGION; i++) {
+    for (ULONG i = 0; i < DETOUR_TRAMPOLINES_PER_REGION; i++) {
         if (pTrampoline[i].pbRemain != NULL &&
             (pTrampoline[i].pbRemain < pbRegionBeg ||
              pTrampoline[i].pbRemain >= pbRegionLim)) {
@@ -1637,7 +1637,7 @@ LONG WINAPI DetourTransactionCommit()
 
 static BYTE detour_align_from_trampoline(PDETOUR_TRAMPOLINE pTrampoline, BYTE obTrampoline)
 {
-    for (LONG n = 0; n < ARRAYSIZE(pTrampoline->rAlign); n++) {
+    for (ULONG n = 0; n < ARRAYSIZE(pTrampoline->rAlign); n++) {
         if (pTrampoline->rAlign[n].obTrampoline == obTrampoline) {
             return pTrampoline->rAlign[n].obTarget;
         }
@@ -1647,7 +1647,7 @@ static BYTE detour_align_from_trampoline(PDETOUR_TRAMPOLINE pTrampoline, BYTE ob
 
 static LONG detour_align_from_target(PDETOUR_TRAMPOLINE pTrampoline, LONG obTarget)
 {
-    for (LONG n = 0; n < ARRAYSIZE(pTrampoline->rAlign); n++) {
+    for (ULONG n = 0; n < ARRAYSIZE(pTrampoline->rAlign); n++) {
         if (pTrampoline->rAlign[n].obTarget == obTarget) {
             return pTrampoline->rAlign[n].obTrampoline;
         }
@@ -2425,7 +2425,7 @@ LONG WINAPI DetourDetach(_Inout_ PVOID *ppPointer,
     //
     LONG cbTarget = pTrampoline->cbRestore;
     PBYTE pbTarget = pTrampoline->pbRemain - cbTarget;
-    if (cbTarget == 0 || cbTarget > sizeof(pTrampoline->rbCode)) {
+    if (cbTarget == 0 || pTrampoline->cbRestore > sizeof(pTrampoline->rbCode)) {
         error = ERROR_INVALID_BLOCK;
         if (s_fIgnoreTooSmall) {
             goto stop;

--- a/src/disasm.cpp
+++ b/src/disasm.cpp
@@ -295,10 +295,10 @@ class CDetourDis
     BOOL                m_bOperandOverride;
     BOOL                m_bAddressOverride;
     BOOL                m_bRaxOverride; // AMD64 only
-    BOOL                m_bVex;
-    BOOL                m_bEvex;
     BOOL                m_bF2;
     BOOL                m_bF3; // x86 only
+    BOOL                m_bVex;
+    BOOL                m_bEvex;
     BYTE                m_nSegmentOverride;
 
     PBYTE *             m_ppbTarget;

